service/github/github.go

```diff
@@ -112,28 +139,57 @@ func (g *PullRequest) postAsReviewComment(ctx context.Context) error {
 		// > 403 You have triggered an abuse detection mechanism and have been
 		// > temporarily blocked from content creation. Please retry your request
 		// > again later.
-		// https://developer.github.com/v3/#abuse-rate-limits
+		// https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limiting
 		if len(comments) >= maxCommentsPerRequest {
 			remaining = append(remaining, c)
 			continue
 		}
 		comments = append(comments, buildDraftReviewComment(c, body))
 	}
+	if err := g.logWriter.Flush(ctx); err != nil {
+		return err
+	}
 
 	if len(comments) == 0 {
 		return nil
 	}
 
+	// send review comments to GitHub.
 	review := &github.PullRequestReviewRequest{
 		CommitID: &g.sha,
 		Event:    github.String("COMMENT"),
 		Comments: comments,
 		Body:     github.String(g.remainingCommentsSummary(remaining)),
 	}
 	_, _, err := g.cli.PullRequests.CreateReview(ctx, g.owner, g.repo, g.pr, review)
+	if err != nil {
+		log.Println(err)
```

Comment:
Can we move this line into the isClient error block and use consistent log format?